circe/src/test/scala/com/kevel/apso/circe/ImplicitsSpec.scala

```diff
@@ -26,6 +26,16 @@ class ImplicitsSpec extends Specification {
 
         res mustEqual expectedJson
       }
+
+      "not throw a StackOverflowError for large lists of paths" in {
+        val paths = (1 to 10000).map(i => (s"a.v$i", i.asJson)).toList
+        fromFullPaths(paths, ".") must not(throwAn[StackOverflowError])
+      }
+
+      "giving precedence to the last path value if duplicate paths exist" in {
+        val res = fromFullPaths(List("a" -> 1.asJson, "a" -> 2.asJson, "a" -> 3.asJson))
+        res mustEqual json"""{"a": 3}"""
```

Comment:
Since we're testing this, can we also add an example in which there are duplicate paths at different depths? Something like the following?

```suggestion
        val json1 = fromFullPaths(List("a" -> 1.asJson, "a" -> 2.asJson, "a" -> 3.asJson))
        json1 mustEqual json"""{"a": 3}"""

        val json2 = fromFullPaths(List("a.b.c" -> 1.asJson, "a.b" -> 2.asJson, "a" -> 3.asJson))
        json2 mustEqual json"""{"a": 3}"""
```